src/Native/System.Native/pal_networking.cpp

```diff
@@ -2338,10 +2344,20 @@ static Error TryChangeSocketEventRegistrationInner(
                0,
                0,
                reinterpret_cast<void*>(data));
+#else
+         EV_SET(&events[i++],
+               static_cast<uint64_t>(socket),
+               EVFILT_READ,
+               (newEvents & PAL_SA_READ) == 0 ? RemoveFlags : AddFlags,
+               0,
+               0,
+               data);
+#endif
```

Comment:
Good idea. I haven't tried it. I will give it a shot.

I however tried ifdef'ing only last line, but clang was giving a warning like: "using macro within a macro has undefined behavior in standard".